syntax = "proto3";

package com.embedd;
option go_package = "golang/core/proto;proto";


enum FileType {
  URL = 0;
  PDF = 1;
  RTF = 2;
  DOC = 3;
  XLS = 4;
  PPT = 5;
  TXT = 6;
  MD = 7;
  // add all supported file that in another document
  // check what with Google docs
};

message ChunkingData {
  // This is the url where the file is located.
  // Based on the chunking type it will be a WEB URL (HTML type)
  // Will be an S3/MINIO link with a proper authentication in case of a file
  string url = 1;
  string site_map = 2;
  bool search_for_sitemap = 3;
  int64  document_id = 4;
  FileType file_type = 5;
}